diff --git a/dom/media/systemservices/PCameras.ipdl b/dom/media/systemservices/PCameras.ipdl
index 123ed52b7270..748f8dc004b3 100644
--- a/dom/media/systemservices/PCameras.ipdl
+++ b/dom/media/systemservices/PCameras.ipdl
@@ -69,7 +69,8 @@ child:
   async ReplyNumberOfCapabilities(int capabilityCount);
   async ReplyAllocateCapture(int streamId);
   async ReplyGetCaptureCapability(VideoCaptureCapability cap);
-  async ReplyGetCaptureDevice(nsCString device_name, nsCString device_id, bool scary);
+  async ReplyGetCaptureDevice(nsCString device_name, nsCString device_id,
+                              bool scary, uint8_t facingMode);
   async ReplyFailure();
   async ReplySuccess();
   async __delete__();
